### DISCUSSION TITLE
Satellites with station keeping

### DISCUSSION THREAD

**Discussion by randomwalkingg** [skyfielders/python-skyfield#725] [Q&A] [answered]
Hi,
Is there a way to propagate satellites that won't drift over time such that their orbit and ground track remain the same. 
Thanks!

**Comment by brandon-rhodes** [comment, marked as answer]
My understanding is that the TLE format is only capable of describing a satellite that's “ballistic” and drifting, and that to simulate a satellite that's doing station keeping, you need a series of TLEs, each applying to a segment of its course between one correction burn and the next.

What ground track do you want the satellite to have? By “ground track to remain the same” do you mean “has the same shape each orbit” or “goes over the same land each orbit / n orbits” or “is geostationary”?

**Reply by randomwalkingg** [reply]
Thank you for your response. Agree with you regarding TLE and it's capability. On the other hand, is there any other way in Skyfield to simulate station-keeping?

As I understand it, station-keeping will force the satellite being in the orbit that was designed for, such that the trajectory of the satellite projected on earth remains the same over time. Basically, as you said, it goes over same land.

**Reply by brandon-rhodes** [reply]
Thanks for clarifying. No, Skyfield does not contain any simulation of either how a satellite would make the decision to fire its thrusters, nor a simulation of what effect that firing have on the TLE orbit. I've done a maneuver like that before in Kerbal Space Program, which simulates the result, but not in software that I myself have written.
